src/main/daml/Daml/Finance/Instrument/Bond/Util.daml

```diff
@@ -59,11 +61,18 @@ createCallableBondClaims dateToTime schedule periodicSchedule useAdjustedDatesFo
     -- claim (in front of the or node, see explanation below).
     principal = give notionalAmount
 
-    -- Currently, it is possible to call on every coupon date
-    callPossibleInit = replicate (length schedule) True
+    -- Find out on which coupon dates it is possible to call the bond.
+    callPossibleInit = map (\s -> isSome $
+      find (\c -> c.adjustedEndDate == s.adjustedEndDate) callableSchedule)
+      schedule
 
-    -- Except on the last one, because that does not make sense (principal repaid in any case)
-    callPossible = init callPossibleInit ++ [False]
+    -- Also search in the other direction, to ensure that no intended call dates are ignored.
```

Comment:
The same function is being written twice, so probably can re-use? Something like `includes: Schedule -> Schedule`?

Reply:
Good point, I have done that now.